controllers/snapshot/snapshot_adapter_test.go

```diff
@@ -1019,6 +1185,149 @@ var _ = Describe("Snapshot Adapter", Ordered, func() {
 			Expect(len(owners) == 1).To(BeTrue())
 			Expect(owners[0].Name).To(Equal(ephemeralEnv.Name))
 		})
+
+		It("ensure ephemeral env and binding will not be created adain when they all exist", func() {
```

Comment:
Should be When statement

Reply:
Thanks, I think it need the same env with previous `It` in `When("An ephemeral Environment environment exists already", func() {`

Reply:
You can have nested whens
```
When("An ephemeral Environment environment exists already") {
    When("Binding exists") {
        It("ensure ephemeral env and binding will not be created again")
    }
}
```

Limitation is, you need to use BeforeEach with nested When

Reply:
Updated. Thanks.